requests/utils.py

```diff
@@ -732,6 +732,24 @@ def to_native_string(string, encoding='ascii'):
 
     return out
 
+# Moved outside of function to avoid recompile every call
+_CLEAN_HEADER_REGEX_BYTE = re.compile(b'^\S[^\\r\\n]*$|^$')
```

Comment:
I think *strictly* we're missing a backslash in this regex. You have `\S`, but I think you want `\\S`.